extensions/resteasy-reactive/rest/runtime/src/test/java/io/quarkus/resteasy/reactive/runtime/mapping/ObservabilityIntegrationRecorderTest.java

```diff
@@ -0,0 +1,50 @@
+package io.quarkus.resteasy.reactive.runtime.mapping;
+
+import io.quarkus.resteasy.reactive.server.runtime.observability.ObservabilityIntegrationRecorder;
+import io.vertx.ext.web.RoutingContext;
+import org.jboss.resteasy.reactive.server.core.Deployment;
+import org.jboss.resteasy.reactive.server.handlers.RestInitialHandler;
+import org.jboss.resteasy.reactive.server.mapping.RequestMapper;
+import org.jboss.resteasy.reactive.server.mapping.URITemplate;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ObservabilityIntegrationRecorderTest {
```

Comment:
I was afraid of this :)

We actually don't want tests like this, because they are super hard to maintain and don't test the actual issue.
What we need is a test `QuarkusUnitTest` (of which we have a huge number already in the codebase) that represents the scenario you found that causes the bug.

Reply:
Yes that would be better. Do you want me to try to write some?

Reply:
If you want to give it a go, that would be nice. Otherwise, I can do it

Reply:
Yeah, apart from the `QuarkusUnitTest`, I would probably skip this `ObservabilityIntegrationRecorderTest` and create one simple one for `PathHelper` because a test like that doesn't need any mocking and you can easily test multiple combinations of paths.

Reply:
@bechhansen any chance you could adjust your PR following @brunobat 's recommendation? It looks like something we want in soonish.

Reply:
Thanks!